### ISSUE TITLE
Typescript error (TS2464): Computed property

### ISSUE THREAD

**Issue by ghost** [emotion-js/emotion#1275]
- `gatsby-plugin-emotion` version: 4.0.6
- `react` version: 16.8.4

**Relevant code:**

```typescript
const StyledLink = styled(Link)({...});

const MobileLinkContainer = styled("div")({
  ... 
  [StyledLink]: {
    padding: "1rem"
  }
});
```

**What happened:**
I get an error: `A computed property name must be of type 'string', 'number', 'symbol', or 'any'. ts(2464)` at the `[StyledLink]` line

**Suggested solution:**
A workaround is to use `[StyledLink as any]`.

**Comment by mwarger** [comment]
I have the same issue and was able to work around it for now by explicitly returning `:any` from the child element.

```javascript
const StyledCheckbox: any = styled.input(({ theme }) => ({}));

const CheckboxLabel = styled.label(({ theme }) => ({
  [StyledCheckbox]: {}
}));
```

**Comment by luk707** [comment]
We've been experiencing the same issue for our projects, Rather than typing the `StyledCheckbox` to any you can do it in the computed property so you keep nice type introspection for the rest of your application:

```ts
const StyledCheckbox = styled.input(({ theme }) => ({}));

const CheckboxLabel = styled.label(({ theme }) => ({
  [StyledCheckbox as any]: {}
}));
```

**Comment by B-Stewart** [comment]
Noticing this still. +1 to get it fixed.

**Comment by Andarist** [comment]
I believe this has been fixed already, but if not - please raise it again. Thanks for understanding.

**Comment by jsonmaur** [comment]
This is still an issue in v10.

**Comment by praneetrohida** [comment]
@Andarist +1 on v10.0.27 with Next.js

**Comment by chadlavi** [comment]
Anyone got anything on this? this is still an issue.

**Comment by Pearce-Ropion** [comment]
Still getting this on version 11.7.1

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

**Comment by srmagura** [comment]
Minimal repro: https://codesandbox.io/s/emotion-1275-w3dkqd?file=/src/App.tsx   It encounters an error at runtime since `@emotion/babel-plugin` is not used, but it does demonstrate the TypeScript compilation error.

**Comment by Andarist** [comment]
> I don't see any way to "fix" this issue. The TypeScript error is totally legit, since TypeScript has no way to know about the transformation applied by @emotion/babel-plugin

Right, so we just assume at the type-level that this transformation is always applied and we add a "fake trait" to our `StyledComponent` interface:
https://github.com/emotion-js/emotion/blob/26e4e3e8b68479f0e3cb8fbec723da47afd6ac98/packages/styled/types/base.d.ts#L41

This allows styled components to be interpolated into our template strings~. But it doesn't allow us to use them as computed properties in the object syntax. We could make it work if we would add `& string` to those types but that could complicate inheritance and "wrapping" (like `styled(styled.div({}))`), so unless somebody would try this change and add good type-level tests to cover it I don't think it's something I would explore myself.

The workaround I usually do to cover this is just this:
```diff
const MobileLinkContainer = styled.div({
-  [StyledLink]: {
+  [String(StyledLink)]: {
    padding: "1rem"
  }
});
```

It's simple, explicit and effective.

**Comment by alexicum** [comment]
```
const MobileLinkContainer = styled.div({
  [`${StyledLink}`]: {
    padding: "1rem"
  }
});
```
